lib/rack/mock.rb

```diff
@@ -182,9 +182,14 @@ class << self
     # Errors
     attr_accessor :errors
 
-    def initialize(status, headers, body, errors = StringIO.new(""))
+    def initialize(status, headers, body, errors = nil)
```

Comment:
I think we should default to `''` instead of `nil`.  This still avoids allocations, since it is a frozen string.  I think it results in the same behavior as well, since StringIO#string returns the input string (which will be the literal frozen string passed to it).